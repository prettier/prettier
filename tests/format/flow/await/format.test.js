runFormatTest(import.meta, ["flow"], {
  errors: { "babel-flow": ["await-keywords.js"] },
});
