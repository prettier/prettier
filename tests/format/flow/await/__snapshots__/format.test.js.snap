// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`await-keywords.js [babel-flow] format 1`] = `
"'await' is only allowed within async functions and at the top levels of modules. (12:7)
  10 |
  11 | export class C2 {
> 12 |   p = await 0;
     |       ^
  13 | }
  14 |
  15 | function foo(promise) { await (promise); }
Cause: 'await' is only allowed within async functions and at the top levels of modules. (12:6)"
`;

exports[`await-keywords.js format 1`] = `
====================================options=====================================
parsers: ["flow"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
export class C1 {
  p = await (0);
}

await (0);

async function foo() {
  function bar(x = await (2)) {}
}

export class C2 {
  p = await 0;
}

function foo(promise) { await (promise); }

function a() {
  return await (1)
}

() => { await (x) };

function foo() {
  await
  (foo);
}

export class C {
  p = await (0);
}

await (0);

=====================================output=====================================
export class C1 {
  p = await(0);
}

await(0);

async function foo() {
  function bar(x = await(2)) {}
}

export class C2 {
  p = await;
  0;
}

function foo(promise) {
  await(promise);
}

function a() {
  return await(1);
}

() => {
  await(x);
};

function foo() {
  await(foo);
}

export class C {
  p = await(0);
}

await(0);

================================================================================
`;
